integration_test/system_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integrationtest
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestKrewSystem(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	test.WithIndex().Krew("install", validPlugin).RunOrFail()
+
+	// needs to be after initial installation
+	prepareOldKrewRoot(test)
+
+	test.Krew("system", "receipts-upgrade").RunOrFailOutput()
+	test.AssertExecutableInPATH("kubectl-" + validPlugin)
+
+	assertReceiptExistsFor(test, validPlugin)
```

Comment:
nit: should this be part of ITest? this way you can do: 

`test.assertReceiptExistsFor(test, validPlugin)` just like  `test.AssertExecutableInPATH("kubectl-" + validPlugin)` 

also note that it's inconsistent one is exported, the other one isn't.

Reply:
That depends on whether we want to keep that assertion method. Also, I would consider receipts more of an implementation detail. It's necessary to assert on them in the current test-case, but should we encourage doing so in other integration tests?

Reply:
in that case  we should add a TODO to all these tests + helpers to remove them after the migration code is scrubbed.